src/components/profile/Profile.js

```diff
@@ -1,51 +1,44 @@
-import React, { Component } from 'react'
-import { connect } from 'react-redux'
-
-import { withRouter } from 'react-router-dom'
+import React from 'react'
 
 import PageContainer from '../common/PageContainer';
 import ProfileDetails from './ProfileDetails'
 import ProfileSection from './ProfileSection'
-import ProfileYourKeys from './ProfileYourKeys'
-import ProfileNotice from './ProfileNotice'
 import ProfileQRCode from './ProfileQRCode';
-
-class Profile extends Component {
-   state = {
-      loader: false
-   }
-
-   componentDidMount() {}
-
-   render() {
-      const { account } = this.props
-
-      return (
-         <PageContainer
-            title={`Account: @${account.accountId ? account.accountId : ``}`}
-         >
-            <ProfileSection>
-               <ProfileDetails account={this.props.account} />
-               { false ?
-               <ProfileYourKeys />
-               : null }
-               { false ?
-               <ProfileNotice />
-               : null }
-               <ProfileQRCode account={this.props.account} />
-            </ProfileSection>
-         </PageContainer>
-      )
-   }
+import { Wallet } from '../../utils/wallet'
+
+const wallet = new Wallet()
+
+export function Profile({ match }) {
+  const { accountId } = match.params
+  const [ loading, setLoading ] = React.useState(true)
```

Comment:
Let's not become more inconsistent with adding React Hooks as well.

For now we have a bit half-assed Redux integration, I'd prefer we clean it up as generally it's been quite useful when debugging complex flows (as Redux Dev tools allows to view state diffs, time-travel, etc).

Reply:
Looks like there are good ways to use Redux together with Hooks as well: https://thoughtbot.com/blog/using-redux-with-react-hooks

Probably worth deeper discussion, as functional style of components looks cleaner.

Reply:
Having worked on several hefty projects with redux and then a few without, I've come down on "redux is probably not worth all the indirection it adds, especially now that React has hooks."

It looks like there's only one piece of global state that we need in this project: the current user. We could make this easily accessible to all components that need it using React's context api. I recently helped implement something similar here: https://github.com/AutarkLabs/open-enterprise/commit/96ea396349208b8e064832caf42a050f3d8340ed#diff-ae42d1e8553b2812cb903dd2add50dd4

As for "state diffs, time-travel, etc" – I have not missed these features at all. I'd wager that the number of complex flows that require debugging can be reduced by removing redux.

Reply:
> It looks like there's only one piece of global state that we need in this project: the current user. 

Ideally we need to keep preloaded data about account globally as well, so that UI is more snappy (i.e. we show cached data before new data is loaded). Stuff like "what message to show after redirect" or "where to redirect after login" is also global state.

Reply:
@marcinbodnar @Patrick1904 what are your thoughts on this? Do you feel like Redux Dev Tools are useful for debugging state changes or not really?

Reply:
Another angle I'd like to consider: I think near-wallet may end up being a great reference project for other NEAR developers building apps with React. To that end, I want to make its code as easy to understand as possible, and as close to current React "best practices" as we can get. My anecdotal impression is that redux is falling out of favor, now that React has better built-in ways to manage data than it did when redux first gained popularity. (A quick search for "redux popularity over time" yielded this as the top result: https://hackernoon.com/goodbye-redux-26e6a27b3a0b.)

Reply:
@chadoh I agree that Redux would be an overkill if all the data is loaded every time.

However we need to make a reference blockchain app and that ideally should account for far slower responses from backend than in normal web app. That means we'll need stuff like caching data locally. Which is generally pretty hard to get working and debug when state can be modified from anywhere.

Reply:
> Do you feel like Redux Dev Tools are useful for debugging state changes or not really?

useful for me, regarding some complex flows we have implemented

Reply:
I'm willing to use redux for now. If I want to re-raise the question of whether it's worth moving away, I'll do so by refactoring one of those hard-to-get-working parts, and see what we think. It's hard to evaluate such architectural alternatives without seeing a good side-by-side comparison.

Reply:
@chadoh I think it's good idea to switch to functional style with hooks when using Redux though:
https://thoughtbot.com/blog/using-redux-with-react-hooks

Reply:
@vgrichina would you like to see this done all at once, or are you ok with it being done piecemeal as we add and edit existing components?

Reply:
@chadoh seems like smth much easier to do piecemeal.

Reply:
@vgrichina I've updated this component to fetch the current user from the redux state, but I doubt we want to add the profile-being-viewed to the global state, since no other components use this data currently. Let me know what you think.